index.js

```diff
@@ -462,18 +461,18 @@ class Funnel extends Plugin {
     throw new Error(`Pattern \`${pattern}\` was not a RegExp, Glob, or Function.`);
   }
 
-  processFile(sourcePath, destPath /*, relativePath */) {
-    this._copy(sourcePath, destPath);
+  processFile(sourcePath, destPath, relativePath) {
+    this._copy(sourcePath, destPath, relativePath);
   }
 
-  _copy(sourcePath, destPath) {
-    let destDir = path.dirname(destPath);
+  _copy(sourcePath, destPath, relativePath) {
+    let destDir = path.dirname(relativePath);
 
     try {
       symlinkOrCopy.sync(sourcePath, destPath);
     } catch (e) {
-      if (!fs.existsSync(destDir)) {
-        fs.mkdirSync(destDir, { recursive: true });
+      if (!this.output.existsSync(destDir)) {
```

Comment:
Does this guard provide any value? I would imagine `fs.mkdirp` + `recursive` does something similar under the hood? Could you investigate?